proto/substrait/algebra.proto

```diff
@@ -594,8 +647,15 @@ message MergeJoinRel {
   RelCommon common = 1;
   Rel left = 2;
   Rel right = 3;
-  repeated Expression.FieldReference left_keys = 4;
-  repeated Expression.FieldReference right_keys = 5;
+  // These fields are deprecated in favor of `keys`.  If they are set then
+  // the two lists (left_keys and right_keys) must have the same length and
+  // the comparion function is considered to be SimpleEqualityType::EQ
+  repeated Expression.FieldReference left_keys = 4 [deprecated = true];
+  repeated Expression.FieldReference right_keys = 5 [deprecated = true];
+  // One or more keys to join on.  The relation is invalid if this is empty.
+  // If a custom comparison function is used then it must be consistent with
+  // the ordering of the input data.
```

Comment:
What does it mean to be consistent with the ordering of the input data?

Reply:
E.g. if your comparison function is < then your data needs to be ordered in ascending order.  If your comparison function is > then your data needs to be ordered in descending order.

In other words, for a given value on the left, as we are walking the right side, the comparison function should only return false once there are no more records that will return true.